torch/distributions/distribution.py

```diff
@@ -71,7 +72,7 @@ def event_shape(self):
         """
         return self._event_shape
 
-    @property
+    @property  # type: ignore[no-redef]
```

Comment:
do we have a mypy git issue link? (is it: https://github.com/python/mypy/issues/6185?)

Reply:
> do we have a mypy git issue link? (is it: [python/mypy#6185](https://github.com/python/mypy/issues/6185)?)

No it is not - this is simply because the name "arg_constraints" is reused as a member variable and a method name at the same time. I don't find a related mypy git issue. This could be resolved by renaming one of them, but that would introduce too much code change IMO.

Reply:
it would be nice to put a comment to explain this

Reply:
Hmm, if arg_constraints is an internal variable, perhaps an underscore prefix should be added to it? ( Which could be done as a separate PR)